app/devUtilities/validate-areas.js

```diff
@@ -0,0 +1,229 @@
+        useUnifiedTopology: true,
+        serverSelectionTimeoutMS: 10000,
+        socketTimeoutMS: 60000,
+        keepAlive: true,
+        keepAliveInitialDelay: 300000,
+    };
+
+    try {
+        await mongoose.connect(mongoURI, configs);
+
+        mongoose.connection.on('disconnected', () => {
+            logStats('🔌  MongoDB connection lost. Attempting to reconnect...');
+            mongoose.connect(mongoURI, configs);
+        });
+
+        mongoose.connection.on('reconnected', () => {
+            logStats('🔄 MongoDB reconnected.');
+        });
+
+        logStats('✅  Connected to MongoDB');
+    } catch (err) {
+        logStats(`❌  Error connecting to MongoDB: ${err}`);
+        process.exit(1);
+    }
+}
+
+// Function to validate Area documents in batches
+async function validateAreasInBatches() {
+    let lastId = null;
+    let hasMore = true;
+
+    // Count the total number of documents in the collection for progress tracking
+    const totalDocumentsCount = await Area.countDocuments();
+    if (totalDocumentsCount === 0) {
+        logMessage('No documents found in the Area collection.');
+        return;
+    }
+
+    // Create a new progress bar instance with ETA and seconds/doc
+    const progressBar = new cliProgress.SingleBar(
+        {
+            format: 'Progress | {bar} | {percentage}% | {value}/{total} Docs | {speed}s/doc | ETA: {eta}s',
+            barCompleteChar: '\u2588',
+            barIncompleteChar: '\u2591',
+            hideCursor: true,
+        },
+        cliProgress.Presets.shades_classic
+    );
+
+    // Start the progress bar
+    progressBar.start(totalDocumentsCount, 0, {
+        speed: 'N/A',
+        eta: 'N/A',
+    });
+
+    // Start time for calculating seconds/doc and ETA
+    const startTime = Date.now();
+
+    while (hasMore) {
+        try {
+            const query = lastId ? { _id: { $gt: lastId } } : {};
+            const areas = await Area.find(query).limit(BATCH_SIZE).sort({ _id: 1 });
+
+            if (areas.length === 0) {
+                hasMore = false;
+                break;
+            }
+
+            for (const area of areas) {
+                totalDocuments++;
+
+                // Check if the document is an Administrative Boundary
+                if (isAdministrativeBoundary(area)) {
+                    totalAdminBoundaries++;
+                }
+
+                if (options.dryrun) {
+                    try {
+                        await area.validate();
+                        logMessage(`💡 [Dry Run] Validated Area: ${area.name || 'Unnamed'} (ID: ${area._id})`);
+                        totalValidated++;
+                    } catch (validationError) {
+                        logMessage(`❌  Validation Error for Area ID: ${area._id} - ${validationError}`);
+                        totalFailures++;
+                    }
+                } else {
+                    try {
+                        await area.save();
```

Comment:
If I understand correctly, just calling `save` will automatically migrate it to the new model?

Reply:
Yes _because_ I [utilized the pre-validate hook](https://github.com/gfw-api/gfw-area/blob/production/app/src/models/area.modelV2.js#L124-L133) provided by Mongoose to add the `adminVersions` collection to an area. Since validation happens before a save, the new code is automatically triggered.